freqtrade/freqtradebot.py

```diff
@@ -299,18 +320,23 @@ def create_trades(self) -> bool:
 
         buycount = 0
         # running get_signal on historical data fetched
-        for _pair in whitelist:
-            if self.strategy.is_pair_locked(_pair):
-                logger.info(f"Pair {_pair} is currently locked.")
+        for pair in whitelist:
+            if self.strategy.is_pair_locked(pair):
+                logger.info(f"Pair {pair} is currently locked.")
                 continue
 
             (buy, sell) = self.strategy.get_signal(
-                _pair, self.strategy.ticker_interval,
-                self.dataprovider.ohlcv(_pair, self.strategy.ticker_interval))
+                pair, self.strategy.ticker_interval,
+                self.dataprovider.ohlcv(pair, self.strategy.ticker_interval))
+
+            if buy and not sell:
+                if not self.get_free_open_trades():
+                    logger.warning("Can't open a new trade: max number of trades is reached")
+                    continue
 
-            if buy and not sell and len(Trade.get_open_trades()) < self.config['max_open_trades']:
-                stake_amount = self._get_trade_stake_amount(_pair)
+                stake_amount = self.get_trade_stake_amount(pair)
                 if not stake_amount:
+                    logger.warning("Stake amount is 0, ignoring trade")
```

Comment:
Not sure about the loglevel of warning ...

we usually use "warning" for things which may need attention  - this sounds more like "info" to me - as it's a valid and probably common possibility to have 0 remaining stake amount available (all stake is tied up in trades).

I would also reword it to `"Stake amount is 0, ignoring possible trade for {pair}."`

Reply:
This one should normally be printed in the meantime since no methods used above to calculate stake_amount can return 0. It was added primarily for mypy satisfaction. However, this can also be useful (or at least be working) taking account upcoming improvements that can be made here.

Anyway, both messages made debug for now to be non-noisy.
Text of this message was adjusted. 👍

Reply:
^^ This one should NOT normally be printed...